src/guiapi/src/display_helper.cpp

```diff
@@ -266,6 +266,108 @@ void render_icon_align(Rect16 rc, const png::Resource *res, color_t clr_back, ic
     }
 }
 
+/**
+ * @brief calculate size (in number of characters) of required rectangle, independent on font
+ * TODO use common algorithm with draw and merge those functions in one
+ *
+ * @param str                           text to be measured
+ * @param max_chars_per_line            max returned number of characters per line
+ * @param ret_numOfUTF8Chars            optional return number of characters (use nullptr if not needed)
+ * @return std::optional<size_ui16_t>   size of needed rectangle, it might be narrower than max_width
+ */
+std::optional<size_ui16_t> characters_meas_text(string_view_utf8 &str, uint16_t max_chars_per_line, uint16_t *numOfUTF8Chars) {
+    if (max_chars_per_line == 0)
+        return std::nullopt;
+
+    std::optional<int> last_space_index = std::nullopt;
+    unichar c = 0;
+
+    int chars_this_line = 0;
+    int chars_longest_line = 0;
+    int chars_tot = 0;
+    int row_no = 0;
+
+    while ((c = str.getUtf8Char()) != 0) {
+        ++chars_tot;
+        switch (c) {
+        case '\n': // new line
+            if (chars_this_line >= chars_longest_line)
+                chars_longest_line = chars_this_line;
+            ++row_no;
+            chars_this_line = 0;
+            break;
```

Comment:
Here should be re-initialized `last_space_index` to 0.

Reply:
done